.github/workflows/lint.yml

```diff
@@ -1,25 +1,32 @@
 name: lint
 
-on: [push, pull_request]
+on:
+  pull_request:
+  push:
+    branches:
+      - main
 
 jobs:
   build:
 
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v1
-    - name: Run pre-commit test
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: 3.x
+    - name: Install pre-commit dependency - terraform-docs
       run: |
-        sudo apt-get install python3-pip gawk
-        curl https://pre-commit.com/install-local.py | python3 -
-        # Terraform Docs
-        curl -Lo ./terraform-docs https://github.com/terraform-docs/terraform-docs/releases/download/v0.9.1/terraform-docs-v0.9.1-$(uname | tr '[:upper:]' '[:lower:]')-amd64
-        chmod +x ./terraform-docs
-        sudo mv ./terraform-docs /usr/bin/terraform-docs
-        # Tflint
-        curl -L "$(curl -s https://api.github.com/repos/terraform-linters/tflint/releases/latest | grep -o -E "https://.+?_linux_amd64.zip")" > tflint.zip && unzip tflint.zip && rm tflint.zip && sudo mv tflint /usr/bin/
-        # Run pre-commit
-        /home/runner/bin/pre-commit --version
-        terraform-docs --version
-        /home/runner/bin/pre-commit run -a
+        TF_DOCS_VER="v0.9.1"
+        pushd /tmp
+          curl -Lo ./terraform-docs https://github.com/terraform-docs/terraform-docs/releases/download/${TF_DOCS_VER}/terraform-docs-${TF_DOCS_VER}-$(uname | tr '[:upper:]' '[:lower:]')-amd64 && chmod +x ./terraform-docs && sudo mv ./terraform-docs /usr/bin/
+        popd
+    - name: Install pre-commit dependency - tflint
+      uses: terraform-linters/setup-tflint@v2
+      with:
+        tflint_version: v0.34.0
```

Comment:
Is there a particular reason why we use this version of tflint?

Reply:
Yes, using the latest one will results in a lot of errors. Will need to update the terraform codes to conform to latest terraform code conventions.
This version is the latest tflint version at the time of the last [commit](https://github.com/HENNGE/terraform-aws-cloudwatch-dashboard/commit/a618422186a3fa62ca8138ff8eaac72873859260) (17 Dec 2021)

Reply:
I see, let's open an issue to update the tflint and terraform code later on